src/main/java/in/projecteka/consentmanager/clients/ClientError.java

```diff
@@ -54,6 +55,16 @@ public static ClientError userNotFound() {
                 new ErrorRepresentation(new Error(USER_NOT_FOUND, "Cannot find the user")));
     }
 
+    public static ClientError transactionPinDidNotMatch() {
+        return new ClientError(UNAUTHORIZED,
+                new ErrorRepresentation(new Error(INVALID_TRANSACTION_PIN, "Invalid transaction pin")));
+    }
+
+    public static ClientError invalidTransactionPin() {
+        return new ClientError(BAD_REQUEST,
+                new ErrorRepresentation(new Error(INVALID_TRANSACTION_PIN, "Invalid transaction pin")));
+    }
+
```

Comment:
These are not client error, I would prefer to keep it inside the user-service.

Reply:
@ganesanarun Because we have a global exception handler, any exception thrown will be handled by that. If it is client error, it will use those error details (such as http status code and error message) to return the response. Else, it will return internal server error as response. If we throw normal exceptions from user service, it will be an internal server error response(instead we want 401 and 400 responses).  Thats why for time being I created these as ClientError type. How can we fix this?

Reply:
I know that, you could have created the clienterror only inside the user service with the comments, or here you could have put a comment, or you could have created common exception in the common folder and wrote a handler tor that :)